guvnor-rest/guvnor-rest-backend/src/main/java/org/guvnor/rest/backend/ProjectResource.java

```diff
@@ -279,7 +279,7 @@ public Response createProject(
             throw new WebApplicationException( Response.status( Response.Status.NOT_FOUND ).entity( repositoryName ).build() );
         }
 
-        Set<Project> projects = projectService.getProjects(repository, repository.getCurrentBranch());
+        Set<Project> projects = projectService.getProjects(repository, "master");
```

Comment:
Not sure. During all this I did find out that we have blind faith on the following
* There is always a master branch 
* There is always a branch
* There is always a repository in a file system
All something we should check as soon as possible.